src/app/core/services/shopping-cart-service/shopping-cart-service.service.ts

```diff
@@ -26,6 +33,18 @@ export class ShoppingCartServiceService {
     return this._selectedProductsIdList.pipe(map(products => products.length));
   }
 
-  constructor() {
+  getSelectedProducts() {
+    const getProductsById = (products) => {
+      return this.selectedProductsIdList.pipe(
+        map(id => {
```

Comment:
Just remark, you can do like this. But not necessary for this case
map(id => products.filter(product => id.includes(product.id)))